src/net.cpp

```diff
@@ -385,6 +385,12 @@ CNode* ConnectNode(CAddress addrConnect, const char *pszDest)
     if (pszDest ? ConnectSocketByName(addrConnect, hSocket, pszDest, Params().GetDefaultPort(), nConnectTimeout, &proxyConnectionFailed) :
                   ConnectSocket(addrConnect, hSocket, nConnectTimeout, &proxyConnectionFailed))
     {
+        if (!IsSelectableSocket(hSocket)) {
+            LogPrintf("Cannot create connection: non-selectable socket created (file descriptor limit exceeded?)\n");
```

Comment:
Well OK @luke-jr is right in that this is not "the" file descriptor limit, just the **select** file descriptor limit. Let's just name the beast as it is:
`Cannot create connection: non-selectable socket created (>=FD_SETSIZE)`